src/js/structs/PodInstance.js

```diff
@@ -0,0 +1,85 @@
+import Item from './Item';
+import PodContainer from './PodContainer';
+import PodInstanceStatus from '../constants/PodInstanceStatus';
+import StringUtil from '../utils/StringUtil';
+
+module.exports = class PodInstance extends Item {
+
+  getContainers() {
+    let containers = this.get('containers') || [];
+    return containers.map((container) => {
+      return new PodContainer(container);
+    });
+  }
+
+  getInstanceStatus() {
+    switch (this.get('status')) {
+      case 'PENDING':
+        return PodInstanceStatus.STAGED;
+
+      case 'STAGING':
+        return PodInstanceStatus.STAGED;
+
+      case 'STABLE':
+        if (this.hasHealthChecks()) {
+          if (this.isHealthy()) {
+            return PodInstanceStatus.HEALTHY;
+          } else {
+            return PodInstanceStatus.UNHEALTHY;
+          }
+        } else {
+          return PodInstanceStatus.RUNNING;
+        }
+
+      case 'DEGRADED':
+        return PodInstanceStatus.UNHEALTHY;
+
+      case 'TERMINAL':
+        return PodInstanceStatus.KILLED;
+
+      default:
+        return Object.assign(Object.create(PodInstanceStatus.NA), {
+          displayName: StringUtil.capitalize(this.get('status').toLowerCase())
+        });
+    }
+  }
+
+  isInStatus(statusArray) {
```

Comment:
If it's regarding only this function, I removed it. It's a leftover after I introduced the `isXXX` tests. If you are referring to the genral `isXXX` test, it feels kind of weird to move them anywhere else, since the sole purpose of the structs is to help in this case.

Your proposal sounds more like `Utility.isXXX( rawData )`, for more functional approach.